pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -155,6 +155,15 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 	if (kc.ShutdownGracePeriod.Duration > 0 || kc.ShutdownGracePeriodCriticalPods.Duration > 0) && !localFeatureGate.Enabled(features.GracefulNodeShutdown) {
 		allErrors = append(allErrors, fmt.Errorf("invalid configuration: Specifying ShutdownGracePeriod or ShutdownGracePeriodCriticalPods requires feature gate GracefulNodeShutdown"))
 	}
+	if localFeatureGate.Enabled(features.PodPriorityBasedGracefulShutdown) {
+		if len(kc.PodPriorityShutdownGracePeriods) != 0 && (kc.ShutdownGracePeriod.Duration > 0 || kc.ShutdownGracePeriodCriticalPods.Duration > 0) {
+			allErrors = append(allErrors, fmt.Errorf("invalid configuration: Cannot specify both PodPriorityShutdownGracePeriods and ShutdownGracePeriod at the same time"))
+		}
+	} else {
+		if len(kc.PodPriorityShutdownGracePeriods) != 0 {
+			allErrors = append(allErrors, fmt.Errorf("invalid configuration: Specifying PodPriorityShutdownGracePeriods requires feature gate PodPriorityBasedGracefulShutdown"))
```

Comment:
nit: for clarity, I would move this check out of the else to avoid unnecessary nesting with else / if

i.e. 

```
if localFeatureGate.Enabled(features.PodPriorityBasedGracefulShutdown) && len(kc.PodPriorityShutdownGracePeriods) != 0 {
    allErrors = append(allErrors, fmt.Errorf("invalid configuration: Specifying PodPriorityShutdownGracePeriods requires feature gate PodPriorityBasedGracefulShutdown"))
}